Exec.TestWithTimeout failed with context.deadlineExceededError

Seen in #5442:

```
START: exec_test.go:44: ExecTestSuite.TestWithTimeout
exec_test.go:47:
    c.Assert(err, IsNil)
... value context.deadlineExceededError = context.deadlineExceededError{} ("context deadline exceeded")
FAIL: exec_test.go:44: ExecTestSuite.TestWithTimeout
OOPS: 4 passed, 1 FAILED
--- FAIL: Test (0.01s)
FAIL
coverage: 29.4% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/command/exec	0.016s
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```

https://github.com/cilium/cilium/blob/3c1c2b6149a3c211f5ae6e55e62cb694ac8202a2/pkg/command/exec/exec_test.go#L44-L47

Maybe we need to dig a little deeper into the `cmd.Start()` implementation to determine how this case gets hit and mitigate it.

I've got it as well
```
PASS
level=debug msg="Starting to watch a prefix" prefix=cilium/state/identities/v1/id revision=166 subsys=kvstore watcher=cilium/state/identities/v1/id
coverage: 12.9% of statements in ./...
ok  	github.com/cilium/cilium/pkg/clustermesh	0.429s	coverage: 12.9% of statements in ./...
?   	github.com/cilium/cilium/pkg/color	[no test files]
=== RUN   Test
START: output_test.go:31: CMDHelpersSuite.TestDumpJSON
{
  "ID": 1,
  "Name": "test"
}
Couldn't parse jsonpath expression: 'unrecognized character in action: U+007B '{''
PASS: output_test.go:31: CMDHelpersSuite.TestDumpJSON	0.000s
OK: 1 passed
--- PASS: Test (0.00s)
PASS
coverage: 58.6% of statements in ./...
ok  	github.com/cilium/cilium/pkg/command	0.007s	coverage: 58.6% of statements in ./...
=== RUN   Test
START: exec_test.go:60: ExecTestSuite.TestCombinedOutput
PASS: exec_test.go:60: ExecTestSuite.TestCombinedOutput	0.002s
START: exec_test.go:68: ExecTestSuite.TestCombinedOutputFailedTimeout
level=error msg="Command execution failed: Timeout" cmd="[sleep inf]" foo=bar
PASS: exec_test.go:68: ExecTestSuite.TestCombinedOutputFailedTimeout	0.002s
START: exec_test.go:52: ExecTestSuite.TestWithCancel
PASS: exec_test.go:52: ExecTestSuite.TestWithCancel	0.001s
START: exec_test.go:96: ExecTestSuite.TestWithFilters
level=error msg="Command execution failed" cmd="[cat /non/existing/file /non/existing/file/filtered/out]" error="exit status 1" foo=bar
level=warning msg="cat: /non/existing/file: No such file or directory" foo=bar
PASS: exec_test.go:96: ExecTestSuite.TestWithFilters	0.002s
START: exec_test.go:44: ExecTestSuite.TestWithTimeout
exec_test.go:47:
    c.Assert(err, IsNil)
... value context.deadlineExceededError = context.deadlineExceededError{} ("context deadline exceeded")
FAIL: exec_test.go:44: ExecTestSuite.TestWithTimeout
OOPS: 4 passed, 1 FAILED
--- FAIL: Test (0.01s)
FAIL
coverage: 29.4% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/command/exec	0.017s
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.

```

A proof that the error message is caused by too short timeouts - after decreasing the timeout to 1 nanosecond I'm getting that error locally every time ;)

```
 mrostecki   fix-test-exec  …  github.com  cilium  cilium  git --no-pager diff
diff --git a/pkg/command/exec/exec_test.go b/pkg/command/exec/exec_test.go
index 646dcddd5..db1285e59 100644
--- a/pkg/command/exec/exec_test.go
+++ b/pkg/command/exec/exec_test.go
@@ -29,7 +29,7 @@ import (
 )
 
 const (
-	timeout = 250 * time.Millisecond
+	timeout = 1 * time.Nanosecond
 )
 
 // Hook up gocheck into the "go test" runner.
 mrostecki   fix-test-exec  …  github.com  cilium  cilium  go test ./pkg/command/exec/
level=error msg="Command execution failed: Timeout" cmd="[sleep inf]" foo=bar
level=error msg="Command execution failed" cmd="[cat /non/existing/file /non/existing/file/filtered/out]" error="exit status 1" foo=bar
level=warning msg="cat: /non/existing/file: No such file or directory" foo=bar

----------------------------------------------------------------------
FAIL: exec_test.go:44: ExecTestSuite.TestWithTimeout

exec_test.go:47:
    c.Assert(err, IsNil)
... value context.deadlineExceededError = context.deadlineExceededError{} ("context deadline exceeded")

OOPS: 4 passed, 1 FAILED
--- FAIL: Test (0.01s)
FAIL
FAIL	github.com/cilium/cilium/pkg/command/exec	0.009s
```

--- a/pkg/command/exec/exec_test.go
+++ b/pkg/command/exec/exec_test.go
@@ -29,7 +29,7 @@ import (
 )
 
 const (
-	timeout = 1 * time.Millisecond
+	timeout = 250 * time.Millisecond
 )
 
 // Hook up gocheck into the "go test" runner.

